src/daft-table/src/lib.rs

```diff
@@ -431,30 +433,52 @@ impl Table {
         agg_expr: &AggExpr,
         groups: Option<&GroupIndices>,
     ) -> DaftResult<Series> {
-        use daft_dsl::AggExpr::*;
         match agg_expr {
-            Count(expr, mode) => Series::count(&self.eval_expression(expr)?, groups, *mode),
-            Sum(expr) => Series::sum(&self.eval_expression(expr)?, groups),
-            ApproxSketch(expr) => Series::approx_sketch(&self.eval_expression(expr)?, groups),
-            ApproxPercentile(ApproxPercentileParams {
-                child: expr,
-                percentiles,
+            &AggExpr::Count(ref expr, mode) => self.eval_expression(expr)?.count(groups, mode),
+            AggExpr::Sum(expr) => self.eval_expression(expr)?.sum(groups),
+            &AggExpr::ApproxPercentile(ApproxPercentileParams {
+                child: ref expr,
+                ref percentiles,
                 force_list_output,
             }) => {
                 let percentiles = percentiles.iter().map(|p| p.0).collect::<Vec<f64>>();
-                Series::approx_sketch(&self.eval_expression(expr)?, groups)?
-                    .sketch_percentile(&percentiles, *force_list_output)
+                self.eval_expression(expr)?
+                    .approx_sketch(groups)?
+                    .sketch_percentile(&percentiles, force_list_output)
             }
-            MergeSketch(expr) => Series::merge_sketch(&self.eval_expression(expr)?, groups),
-            Mean(expr) => Series::mean(&self.eval_expression(expr)?, groups),
-            Min(expr) => Series::min(&self.eval_expression(expr)?, groups),
-            Max(expr) => Series::max(&self.eval_expression(expr)?, groups),
-            AnyValue(expr, ignore_nulls) => {
-                Series::any_value(&self.eval_expression(expr)?, groups, *ignore_nulls)
+            AggExpr::ApproxCountDistinct(expr) => self
+                .eval_expression(expr)?
+                .hash(None)?
+                .into_series()
```

Comment:
We probably can implement `approx_count_distinct` on the `UInt64Array` directly since it's the only type supported by this anyways. We can probably skip it being a Trait if its only 1 impl as well.

Reply:
Updated!